test/e2e/upgrade_test.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package e2e
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+)
+
+const (
+	upgradeNamespace = "upgrade-workload"
+)
+
+// Test backup and restore of Kibishi using restic
+var _ = Describe("[Upgrade][Restic] Velero upgrade tests on cluster using the plugin provider for object storage and Restic for volume backups", backup_upgrade_restore_with_restic)
+
+var _ = Describe("[Upgrade][Snapshot] Velero upgrade tests on cluster using the plugin provider for object storage and snapshots for volume backups", backup_upgrade_restore_with_snapshots)
+
+func backup_upgrade_restore_with_snapshots() {
+	backup_upgrade_restore_test(true)
+}
+
+func backup_upgrade_restore_with_restic() {
+	backup_upgrade_restore_test(false)
+}
+
+func backup_upgrade_restore_test(useVolumeSnapshots bool) {
+	var (
+		backupName, restoreName string
+	)
+	upgradeFromVeleroCLI := upgradeFromVeleroCLI
+
+	client, err := newTestClient()
+	Expect(err).To(Succeed(), "Failed to instantiate cluster client for backup tests")
+
+	BeforeEach(func() {
+		if (len(upgradeFromVeleroVersion)) == 0 {
+			Skip("An original velero version is required to run upgrade test, please run test with upgrade-from-velero-version=<version>")
+		}
+		if useVolumeSnapshots && cloudProvider == "kind" {
+			Skip("Volume snapshots not supported on kind")
+		}
+		//Assume velero server image tag is identical to version
+		//Download velero CLI if it's empty
+		if (len(upgradeFromVeleroCLI)) == 0 {
+			upgradeFromVeleroCLI, err = installVeleroCLI(upgradeFromVeleroVersion)
+			Expect(err).To(Succeed())
+		}
+
+		var err error
+		flag.Parse()
+		uuidgen, err = uuid.NewRandom()
+		Expect(err).To(Succeed())
+		if installVelero {
```

Comment:
If `veleroInstall` is not set to `true`, we should skip this test. Members of the team and community run the E2E tests against existing installs so we shouldn't modify those installations.

Reply:
That makes sense a lot, the existed installation should not be modified, so upgrade test will be skipped, thanks.